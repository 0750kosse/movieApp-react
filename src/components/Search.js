import React from 'react';

function Search({ moviesSearch, handleChange, handleSubmit, ratingColours }) {
  const url = "http://image.tmdb.org/t/p/w185/"
  const movieList = moviesSearch.map(movie => {
    const { id, poster_path, title, vote_average, overview } = movie
    return (
      <div key={id} className="movie-card">
        <img src={url + poster_path} alt="poster" className="movie-poster"></img>
        <p className="movie-title">{title}</p>
        <p className={`movie-rating ${ratingColours(vote_average)}`}>{vote_average}</p>
        <div className="movie-overview">
          <h2>Overview</h2>
          <p>{overview}</p></div>
      </div>
    )
  })
  return (
    <div className="search-container">
      <div className="form-container">
        <form onSubmit={handleSubmit}>
          <input type="text" placeholder="Search movie" name="searched" onChange={handleChange}></input>
        </form>
      </div>
      <div className="movies-container">
        {movieList}
      </div>
    </div>
  )
}

export default Search;