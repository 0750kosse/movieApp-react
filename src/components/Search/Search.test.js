import React from 'react';
import Search from './Search'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';

const movieList = [{
  id: 2,
  poster_path: '/sdakldjfal',
  title: 'batman',
  vote_average: 9,
  overview: 'batman and robin'
},
{
  id: 4,
  poster_path: '/2asad',
  title: 'tarzan',
  vote_average: 12,
  overview: 'tarzan and jane'
}]
const setVoteColor = jest.fn

const searchedMovs = [
  { title: 'batman', id: '3' },
  { title: 'spiderman', id: '9' }]

describe('Search component', () => {
  it('matches the snapshot', () => {
    const wrapper = renderer.create(<Search
      moviesSearch={movieList}
      ratingColours={setVoteColor} />).toJSON();
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly when items are passed in', () => {
    const wrapper = shallow(<Search
      moviesSearch={searchedMovs}
      ratingColours={setVoteColor} />)
    expect(wrapper).toMatchSnapshot()
  })
})
