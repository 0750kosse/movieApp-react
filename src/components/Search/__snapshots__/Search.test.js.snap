// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches the snapshot 1`] = `
<div
  className="search-container"
>
  <div
    className="form-container"
  >
    <form>
      <input
        name="searched"
        placeholder="Search movie"
        type="text"
      />
    </form>
  </div>
  <div
    className="movies-container"
  >
    <div
      className="movie-card"
    >
      <img
        alt="poster"
        className="movie-poster"
        src="http://image.tmdb.org/t/p/w185//sdakldjfal"
      />
      <p
        className="movie-title"
      >
        batman
      </p>
      <p
        className="movie-rating function mockConstructor() {
        return fn.apply(this, arguments);
      }"
      >
        9
      </p>
      <div
        className="movie-overview"
      >
        <h2>
          Overview
        </h2>
        <p>
          batman and robin
        </p>
      </div>
    </div>
    <div
      className="movie-card"
    >
      <img
        alt="poster"
        className="movie-poster"
        src="http://image.tmdb.org/t/p/w185//2asad"
      />
      <p
        className="movie-title"
      >
        tarzan
      </p>
      <p
        className="movie-rating function mockConstructor() {
        return fn.apply(this, arguments);
      }"
      >
        12
      </p>
      <div
        className="movie-overview"
      >
        <h2>
          Overview
        </h2>
        <p>
          tarzan and jane
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search component renders correctly when items are passed in 1`] = `ShallowWrapper {}`;
